src/Microsoft.NetCore.Analyzers/Core/ImmutableCollections/DoNotCallToImmutableCollectionOnAnImmutableCollectionValue.Fixer.cs

```diff
@@ -19,10 +26,70 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        private static readonly ImmutableArray<string> ToImmutableMethodNames = new[]
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            "ToImmutableArray",
+            "ToImmutableDictionary",
+            "ToImmutableHashSet",
+            "ToImmutableList",
+            "ToImmutableSortedDictionary",
+            "ToImmutableSortedSet"
+        }.ToImmutableArray();
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Diagnostic diagnostic = context.Diagnostics.FirstOrDefault();
+            if (diagnostic == null)
+            {
+                return;
+            }
+
+            Document document = context.Document;
```

Comment:
I think the style of this repo is "prefer var over explicit type".

Reply:
@genlu sure? I mean, I prefer using vars too, but the reason I went with explicit types is that other analyzers and fixers were written that way...

Reply:
You can actually see the style from the editorconfig file, which is "use var everywhere" :)
https://github.com/dotnet/roslyn-analyzers/blob/master/.editorconfig#L59

Reply:
There might be some inconsistency in the source, we will need to run the code cleanup tools to fix those, but it's not in the scope of this PR.